Pointers/SharedPtr.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <iostream>
+#include "Counter.h"
+
+using std::cout;
+using std::endl;
```

Comment:
Don't have `using` statements in headers (especially not in the global namespace), it pollutes the namespace for others